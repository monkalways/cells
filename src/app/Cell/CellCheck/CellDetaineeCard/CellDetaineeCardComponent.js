import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  cellCheck: PropTypes.shape({
    visual: PropTypes.bool.isRequired,
    verbal: PropTypes.bool.isRequired,
  }),
  onVisualClick: PropTypes.func,
  onVerbalClick: PropTypes.func,
};

const defaultProps = {
  cellCheck: null,
  onVisualClick: null,
  onVerbalClick: null,
};

export const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
  cellCheck,
  onVisualClick,
  onVerbalClick,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter
      cellCheck={cellCheck}
      onVisualClick={onVisualClick}
      onVerbalClick={onVerbalClick}
    />
  </Card>
);

CellDetaineeCardComponent.propTypes = propTypes;
CellDetaineeCardComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(CellDetaineeCardComponent);
