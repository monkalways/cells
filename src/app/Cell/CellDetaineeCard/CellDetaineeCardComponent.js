import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, withStyles } from '@material-ui/core';

import CellDetaineeCardHeader from './CellDetaineeCardHeader';
import CellDetaineeCardFooter from './CellDetaineeCardFooter';
import detaineeImage from './detainee.PNG';
import detaineeAnonymousImage from './detainee-anonymous.PNG';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  isAuthenticated,
}) => {
  const getDetaineeImage = () => {
    if (isAuthenticated) {
      const { intakePhotoResourceUri } = cellDetainee;
      if (intakePhotoResourceUri) {
        return intakePhotoResourceUri;
      }
      return detaineeImage;
    }
    return detaineeAnonymousImage;
  };
  return (
    <Card className={classes.card}>
      <CellDetaineeCardHeader cellDetainee={cellDetainee} />
      <CardMedia
        className={classes.media}
        image={getDetaineeImage()}
        title={`${cellDetainee.firstName} ${cellDetainee.lastName}`}
      />
      <CellDetaineeCardFooter
        cellDetainee={cellDetainee}
        isAuthenticated={isAuthenticated}
      />
    </Card>
  );
};

CellDetaineeCardComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: '#EBEBEB',
  },
  media: {
    height: 0,
    paddingTop: '66.25%', // 16:9
  },
}))(CellDetaineeCardComponent);
