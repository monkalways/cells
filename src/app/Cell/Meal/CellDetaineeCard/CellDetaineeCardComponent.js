import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  meal: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  meal: null,
};

export const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
  meal,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter
      meal={meal}
      onAcceptClick={onAcceptClick}
      onRejectClick={onRejectClick}
      onNotApplicableClick={onNotApplicableClick}
    />
  </Card>
);

CellDetaineeCardComponent.propTypes = propTypes;
CellDetaineeCardComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(CellDetaineeCardComponent);
