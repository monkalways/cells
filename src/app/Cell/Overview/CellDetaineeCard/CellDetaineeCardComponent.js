import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

const OverviewCellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter cellDetainee={cellDetainee} isAuthenticated={isAuthenticated} />
  </Card>
);

OverviewCellDetaineeCardComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(OverviewCellDetaineeCardComponent);
