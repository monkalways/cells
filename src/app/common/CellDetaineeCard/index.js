export { default as CellDetaineeCardHeader } from './CellDetaineeCardHeader';
export { default as CellDetaineeCardContent } from './CellDetaineeCardContent';
