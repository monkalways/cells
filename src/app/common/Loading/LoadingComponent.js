import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
