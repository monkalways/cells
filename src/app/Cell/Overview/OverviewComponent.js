import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';

import CellDetaineeGrid from '../CellDetaineeGrid';
import OverviewCellDetaineeCard from './OverviewCellDetaineeCard';
import OverviewFooter from './OverviewFooter';
import Loading from '../../common/Loading';

const propTypes = {
  cellDetainees: PropTypes.arrayOf(PropTypes.shape({})),
  isCellDetaineesLoaded: PropTypes.bool.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  getCellDetainees: PropTypes.func.isRequired,
  handleSignIn: PropTypes.func.isRequired,
  cellName: PropTypes.string.isRequired,
};

const defaultProps = {
  cellDetainees: [],
};

class OverviewComponent extends Component {
  componentDidMount() {
    const { cellName, getCellDetainees } = this.props;
    getCellDetainees(cellName);
  }

  render() {
    const {
      cellDetainees,
      isCellDetaineesLoaded,
      isAuthenticated,
      handleSignIn,
      cellName,
    } = this.props;
    return (
      <React.Fragment>
        <CellDetaineeGrid>
          {isCellDetaineesLoaded ? (
            <React.Fragment>
              {cellDetainees.map((cellDetainee) => (
                <Grid key={cellDetainee.id} item sm={4}>
                  <OverviewCellDetaineeCard
                    cellDetainee={cellDetainee}
                    isAuthenticated={isAuthenticated}
                  />
                </Grid>
              ))}
            </React.Fragment>
          ) : (
            <Loading />
          )}
        </CellDetaineeGrid>
        <OverviewFooter
          isAuthenticated={isAuthenticated}
          onSignIn={handleSignIn}
          cellName={cellName}
        />
      </React.Fragment>
    );
  }
}

OverviewComponent.propTypes = propTypes;
OverviewComponent.defaultProps = defaultProps;

export default OverviewComponent;
