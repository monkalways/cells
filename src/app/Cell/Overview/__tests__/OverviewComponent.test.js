import React from 'react';
import { shallow } from 'enzyme';

import { OverviewComponent } from '../OverviewComponent';
import Loading from '../../../common/Loading';
import CellDetaineeCard from '../CellDetaineeCard';

describe('OverviewComponent', () => {
  let cellDetainees;
  let isCellDetaineesLoaded;
  let isAuthenticated;
  let cellName;
  let getCellDetainees;
  let handleSignIn;

  beforeEach(() => {
    cellDetainees = [];
    isCellDetaineesLoaded = false;
    isAuthenticated = false;
    cellName = 'c1';
    getCellDetainees = jest.fn();
    handleSignIn = jest.fn();
  });

  const setup = () => shallow(<OverviewComponent
    cellDetainees={cellDetainees}
    isCellDetaineesLoaded={isCellDetaineesLoaded}
    isAuthenticated={isAuthenticated}
    cellName={cellName}
    getCellDetainees={getCellDetainees}
    handleSignIn={handleSignIn}
  />);

  it('should render cell detainee cards when cell detainees are loaded', () => {
    cellDetainees = [
      {
        id: '123',
      },
      {
        id: '456',
      },
    ];
    isCellDetaineesLoaded = true;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CellDetaineeCard)).toExist();
  });

  it('should render Loading component when cell detainees are not loaded', () => {
    isCellDetaineesLoaded = false;
    const wrapper = setup();
    expect(wrapper.find(Loading)).toExist();
  });

  it('should get cell detainees during componentDidMount', () => {
    const wrapper = setup();
    wrapper.instance().componentDidMount();

    expect(getCellDetainees).toBeCalledWith(cellName);
  });
});
