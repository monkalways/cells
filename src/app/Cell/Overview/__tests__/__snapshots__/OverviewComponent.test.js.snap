// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewComponent should render cell detainee cards when cell detainees are loaded 1`] = `
<Fragment>
  <WithStyles(CellDetaineeGridComponent)>
    <WithStyles(Grid)
      item={true}
      key="123"
      sm={4}
    >
      <WithStyles(CellDetaineeCardComponent)
        cellDetainee={
          Object {
            "id": "123",
          }
        }
        cellName="c1"
        isAuthenticated={false}
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="456"
      sm={4}
    >
      <WithStyles(CellDetaineeCardComponent)
        cellDetainee={
          Object {
            "id": "456",
          }
        }
        cellName="c1"
        isAuthenticated={false}
      />
    </WithStyles(Grid)>
  </WithStyles(CellDetaineeGridComponent)>
  <WithStyles(withRouter(OverviewFooterComponent))
    cellName="c1"
    isAuthenticated={false}
    onSignIn={[MockFunction]}
  />
</Fragment>
`;
